### ISSUE TITLE
Catmandu::Store  default_plugins configuration needs also default_options

### ISSUE THREAD

**Issue by phochste** [LibreCat/Catmandu#323]
I need to create a default_plugin for a store that is configurable. E.g. the `catmandu.yml`:

```(yaml)
---
store:
  main:
    package: DBI
    options:
      default_plugins: [ 'FooBar']
      default_options:
             foobar_be_nice: 1
             foobar_colors:
                     - red
                     - green
      bags:
            bag1:  ...
            bag2: ...
            bag3: ...
```

This is now not possible. I have to copy the `foobar_be_nice` and `foobar_colors` to every bag.

**Comment by vpeil** [comment]
why not use references within the YAML file?

**Comment by phochste** [comment]
@vpeil because not all bags are always enumerated in a Catmandu configuration files. E.g. the `Versioning` plugins creates extra `_version` bags that a usually not listed in a configuration file.

**Comment by nics** [comment]
implemented in d9859b52b3cc3d82bc453387a40e613cc9505f00
